pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+	if err != nil {
+		klog.Errorf("unable to manage manager cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleBindingMetrics(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage metrics cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleBindingProxy(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage proxy cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRole(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleMonitoring(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleBinding(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleBindingMonitoring(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageConfigmap(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage configmap err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageCustomResourceDefinition(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage leaderworkerset CRD err: %v", err)
+		return err
+	}
+
+	deployment, _, err := c.manageDeployments(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage deployment err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageServiceAccount(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service account err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageServiceController(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageServiceWebhook(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageMutatingWebhook(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageValidatingWebhook(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service err: %v", err)
+		return err
+	}
+
+	_, err = c.manageServiceMonitor(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage service account err: %v", err)
+		return err
+	}
+
+	_, _, err = v1helpers.UpdateStatus(c.ctx, c.leaderWorkerSetOperatorClient, func(status *operatorv1.OperatorStatus) error {
+		resourcemerge.SetDeploymentGeneration(&status.Generations, deployment)
+		return nil
+	})
+
+	return err
+}
+
+func (c *TargetConfigReconciler) manageConfigmap(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*v1.ConfigMap, bool, error) {
```

Comment:
some of these can be done via NewStaticResourceController

Reply:
Agree. But since we are waiting for your PR in library-go to merge, I didn't change. Once it merges, I'll open followup PR to migrate some resources to StaticResourceController.